tools/ARIAtools/extractProduct.py

```diff
@@ -407,11 +453,42 @@ def export_products(full_product_dict, bbox_file, prods_TOTbbox, layers, dem=Non
                     outFileUnw=os.path.join(outDir,'unwrappedPhase',product_dict[1][i[0]][0])
                     outFileConnComp=os.path.join(outDir,'connectedComponents',product_dict[1][i[0]][0])
 
+                    # change outputFormat if resampling is specified
+                    if arrshape is not None and outputFormat=='VRT':
```

Comment:
@rzinke There should be a way to keep  VRT even when downsampling. Similar to what we do for the coherence VRT having a suffix "full" for the uncropped verison and the cropped being just the date pairs. We definitely want to avoid the need to generate ENVI file if possible.@sssangha what do you think.

You could also leave the full being the uncropped full resolution version, and have the cropped and the downsampled being the regular date pair combination.

Reply:
Hi @dbekaert and @sssangha, I have added the suffix "resamp" to the end of the downsampled VRT files to avoid necessarily writing to ENVI format. Overwriting the VRTs produces a RuntimeError that is ignored: 

Exception ignored in: <built-in function delete_Dataset>
RuntimeError: /Users/rzinke/Documents/Tibet/Dsample/Test/connectedComponents/20180719_20180625resamp, band 1: WriteBlock() not supported for this dataset.

A quick search for "WriteBlock() not supported for this dataset" turns up this page: https://stackoverflow.com/questions/10635107/how-to-write-manipulated-raster-values-to-ascii-grid-with-gdal which suggests writing to a different filetype (this error did not appear with the ENVI output format enforced). Seeing as this is not a fatal error, however, and that the routine seems to produce valid timeseries results, perhaps we can ignore it? 

Here is how I have currently rewritten the code: 
                    if arrshape:
                        print('Resampling unwrapped phase: {}'.format(outFileUnw))
                        # resample
                        gdal.Warp(outFileUnw+'resamp', outFileUnw, options=gdal.WarpOptions(format=outputFormat, cutlineDSName=prods_TOTbbox, outputBounds=bounds, width=arrshape[1], height=arrshape[0], resampleAlg='lanczos',multithread=True, options=['NUM_THREADS=%s'%(num_threads)+' -overwrite']))
                        # update VRT
                        gdal.BuildVRT(outFileUnw+'.vrt', outFileUnw+'resamp', options=gdal.BuildVRTOptions(options=['-overwrite']))